soh/soh/OTRGlobals.cpp

```diff
@@ -1509,3 +1453,89 @@ extern "C" s32 Randomizer_GetRandomizedItemId(GetItemID ogId, s16 actorId, s16 a
+    const int maxBufferSize = sizeof(font->msgBuf);
+    CustomMessageEntry messageEntry;
+    if (gSaveContext.n64ddFlag) {
+        if (textId == TEXT_RANDOMIZER_GI_ESCAPE_HATCH) {
+            messageEntry =
+                Randomizer_GetCustomGetItemMessage((GetItemID)GET_PLAYER(globalCtx)->getItemId, buffer, maxBufferSize);
+        } else if (textId == 0x2053 && Randomizer_GetSettingValue(RSK_GOSSIP_STONE_HINTS) != 0 &&
+            (Randomizer_GetSettingValue(RSK_GOSSIP_STONE_HINTS) == 1 ||
+             (Randomizer_GetSettingValue(RSK_GOSSIP_STONE_HINTS) == 2 &&
+              Player_GetMask(globalCtx) == PLAYER_MASK_TRUTH) ||
+             (Randomizer_GetSettingValue(RSK_GOSSIP_STONE_HINTS) == 3 && CHECK_QUEST_ITEM(QUEST_STONE_OF_AGONY)))) {
+
+            s16 actorParams = msgCtx->talkActor->params;
+
+            // if we're in a generic grotto
+            if (globalCtx->sceneNum == 62 && actorParams == 14360) {
+                // look for the chest in the actorlist to determine
+                // which grotto we're in
+                int numOfActorLists =
+                    sizeof(globalCtx->actorCtx.actorLists) / sizeof(globalCtx->actorCtx.actorLists[0]);
+                for (int i = 0; i < numOfActorLists; i++) {
+                    if (globalCtx->actorCtx.actorLists[i].length) {
+                        if (globalCtx->actorCtx.actorLists[i].head->id == 10) {
+                            // set the params for the hint check to be negative chest params
+                            actorParams = 0 - globalCtx->actorCtx.actorLists[i].head->params;
+                        }
+                    }
+                }
+            }
+
+            RandomizerCheck hintCheck =
+                Randomizer_GetCheckFromActor(globalCtx->sceneNum, msgCtx->talkActor->id, actorParams);
+
+            messageEntry = Randomizer_CopyHintFromCheck(hintCheck);
+        } else if (textId == TEXT_ALTAR_CHILD || textId == TEXT_ALTAR_ADULT) {
+            // rando hints at altar
+            messageEntry = Randomizer_CopyAltarMessage();
+        } else if (textId == TEXT_GANONDORF) {
+            if (INV_CONTENT(ITEM_ARROW_LIGHT) == ITEM_ARROW_LIGHT) {
+                messageEntry = Randomizer_CopyGanonText();
+            } else {
+                messageEntry = Randomizer_CopyGanonHintText();
+            }
+        } else if (textId == TEXT_SCRUB_POH || textId == TEXT_SCRUB_STICK_UPGRADE || textId == TEXT_SCRUB_NUT_UPGRADE) {
+            messageEntry = Randomizer_CopyScrubMessage(textId);
+        }
+    }
+    if (textId == TEXT_GS_NO_FREEZE || textId == TEXT_GS_FREEZE) {
+        if (CVar_GetS32("gInjectSkulltulaCount", 0) != 0) {
+            if (CVar_GetS32("gSkulltulaFreeze", 0) != 0) {
+                textId = TEXT_GS_NO_FREEZE;
+            } else {
+                textId = TEXT_GS_FREEZE;
+            }
+            messageEntry = CustomMessageManager::Instance->RetrieveMessage("BaseGameOverrides", textId);
```

Comment:
i'm not sure how i feel about strings for tableIds, but i can see using strings for tableId as a way to enable easier custom tables with modding in the future.

might be good to put the "always there" type strings into an enum or defines or something in `CustomMessageTypes.h` to avoid needing to update string literals in multiple places if we change our minds about how to name something

Reply:
I do already have something in place for rando at least, with static variables on the Randomizer class holding the IDs for its message tables. I think I might have just forgotten to replace this one after verifying it worked.